File: mobile/lib/features/payment/data/models/payment.dart
/// Correspond au DTO PaymentResponse.
class Payment {
  Payment({
    required this.id,
    required this.shipmentRequestId,
    required this.offerId,
    required this.amountXaf,
    required this.provider,
    required this.status,
    this.providerReference,
  });

  final String id;
  final String shipmentRequestId;
  final String offerId;
  final double amountXaf;
  final String provider;
  final String? providerReference;
  final String status; // EN_ATTENTE, REUSSI, ECHOUE

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        id: json['id'],
        shipmentRequestId: json['shipmentRequestId'],
        offerId: json['offerId'],
        amountXaf: (json['amountXaf'] as num).toDouble(),
        provider: json['provider'],
        providerReference: json['providerReference'],
        status: json['status'],
      );
}

File: backend/src/main/java/com/fretcorridor/backend/payment/entity/Payment.java
package com.fretcorridor.backend.payment.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "payments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    @Id
    @GeneratedValue
    private UUID id;

    @Column(name = "shipment_request_id", nullable = false)
    private UUID shipmentRequestId;

    @Column(name = "offer_id", nullable = false)
    private UUID offerId;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "amount_xaf", nullable = false, precision = 12, scale = 2)
    private BigDecimal amountXaf;

    @Column(nullable = false, length = 30)
    @Builder.Default
    private String provider = "SIMULATED";

    @Column(name = "provider_reference", length = 100)
    private String providerReference;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private PaymentStatus status = PaymentStatus.EN_ATTENTE;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        Instant now = Instant.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = Instant.now();
    }
}
